### ISSUE TITLE
The selection of error "e"

### ISSUE THREAD

**Issue by yhy258** [lightonai/principled-dfa-training#3]
Thank you for sharing your code.

I am writing to ask why you set the error term e as \hat y - y, even if you set the criterion as Cross Entropy Loss.
To my knowledge, the reason for setting the error as \hat y - y is to assume the loss function is MSE.

**Comment by yhy258** [comment]
I just realized if I set e = dL/da_y and assume employing cross entropy loss with softmax activation function, then the above claim is satisfied.
